Add more detail docs about using `querier.max-concurrent`. (#3900)
grafana/loki: docs/sources/configuration/_index.md

# CLI flag: -querier.query-ingesters-within
[query_ingesters_within: <duration> | default = 0s]

# The maximum number of top-level LogQL queries that will execute at the same time, per querier process.
# If using the query frontend, this should be set to at least (`-querier.worker-parallelism` * `number of query frontend replicas`).

> # Otherwise queries may queue in the queriers and not the frontend, which will affect QoS.

Reviewer: This isn't quite correct. Really it should be set to match the number of CPU cores available on the querier (or slightly less than the number of cores if you are in a shared environment like we are)

For example: running one querier with 40 worker-parallelism won't gain you much if you only have 4 cores

Ideally it should be set to match the machine cores and then you should run enough queriers to match (worker-parallelism * frontend replicas)/max-concurrent

Author: @slim-bean Updated the doc!

# CLI flag: -querier.query-ingesters-within
[query_ingesters_within: <duration> | default = 0s]

# The maximum number of top-level LogQL queries that will execute at the same time, per querier process.

> # It should be set to match the number of CPU cores available on the querier (or slightly less than the number of cores if you are in a shared environment).
> # If using the query frontend, Then you should run enough queriers to match (`-querier.worker-parallelism` * `number of query frontend replicas`)/`-querier.max-concurrent`.
> # Otherwise queries may queue in the query frontend, which will affect QoS.

Reviewer 2: This suggestion is meant to improve readability and grammar.  I removed hyphens from the formula given for quantity of queriers.  They looked like negative signs to me, and since there was one in the numerator and one in the denominator which cancel each other out. I don't think you meant them as negative signs anyway.

```suggestion
# Set it to match the number of CPU cores available on the querier. Set it to slightly less than the number of cores if you are in a shared environment.
# If using the query front end, then run enough queriers to match: `(querier.worker-parallelism * number of query front end replicas) / querier.max-concurrent`.
# Without enough queriers, queries may queue in the query front end, which will affect the quality of service.
```

Author: Thanks @KMiller-Grafana for the feedback. 
1. The minus sign(`-`) is actually part of the CLI flag name (e.g: `-querier.worker-paralleism` is the full name you see when you work with LogCLI. So we use it the same here.
2.  I would prefer to keep `frontend` rather than `front end`. Because its name of the component in Loki. (`Query frontend`

I agree with other changes :) Thanks!

> # If using the query frontend, then you should run enough queriers to match (`-querier.worker-parallelism` * `number of query frontend replicas`)/`-querier.max-concurrent`.
> # Without enough queriers, queries may queue in the query frontend, which will affect the quality of service.
> # Alternatively, consider using `-querier.worker-match-max-concurrent` to force worker parallelism to match `-querier.max-concurrent`.

Reviewer 3: ```suggestion
# If using the query frontend, this should be set to at least (-querier.worker-parallelism * number of query frontend replicas).
# Otherwise queries may queue in the queriers and not the frontend, which will affect QoS.
# Alternatively, consider using -querier.worker-match-max-concurrent to force worker parallelism to match -querier.max-concurrent.
```
